Libraries/LibWeb/IndexedDB/Internal/Algorithms.cpp

```diff
@@ -1213,4 +1213,24 @@ GC::Ref<IDBRequest> asynchronously_execute_a_request(JS::Realm& realm, IDBReques
     return request;
 }
 
+// https://w3c.github.io/IndexedDB/#generate-a-key
+WebIDL::ExceptionOr<u64> generate_a_key(JS::Realm& realm, GC::Ref<ObjectStore> store)
+{
+    // 1. Let generator be store’s key generator.
+    auto generator = store->key_generator().value();
+
+    // 2. Let key be generator’s current number.
+    auto key = generator.current_number();
+
+    // 3. If key is greater than 2^53 (9007199254740992), then return failure.
+    if (key > 9007199254740992)
```

Comment:
Don't we have a JS::MAX_SAFE_INT constant or some such for this?

Reply:
There is a `MAX_SAFE_INTEGER_VALUE`, but that is set to `2^53 - 1` (one away from the spec value)
Something like this?

```suggestion
    if (key > MAX_SAFE_INTEGER_VALUE + 1)
```

Reply:
eh. yeah I suppose we could do that. we could also munge the expression. 

`if (!(MAX_SAFE_INTEGER_VALUE <= key)`. 

that doesn't feel that much easier to read than the + 1 though.